File: GestionStagiaires/src/main/java/com/GestionStagiaires/GestionStagiaires/Model/Absence.java
package com.GestionStagiaires.GestionStagiaires.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.Date;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Absence {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Date dateAbsence;

    @ManyToOne
    @JoinColumn(name = "stagiaire_id")
    private Stagiaire stagiaire;

    @ManyToOne
    @JoinColumn(name = "stage_id")
    private Stage stage;

    public Absence(Date dateAbsence, Stagiaire stagiaire) {
        this.dateAbsence = dateAbsence;
        this.stagiaire = stagiaire;
    }
}

File: gestionStagiares_Front/src/app/Model/absence.ts
// absence.model.ts

import { Stage } from "./stage";
import { Stagiaire } from "./stagiaire";

export class Absence {
  id!: number;
  dateAbsence!: Date;
  stagiaire!: Stagiaire;

  constructor(dateAbsence: Date, stagiaire: Stagiaire) {
    this.dateAbsence = dateAbsence;
    this.stagiaire = stagiaire;
  }
}
